src/pages/api/revalidate-all-tutorials.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import type { NextApiRequest, NextApiResponse } from 'next/types'
+import { StatusCodes } from 'http-status-codes'
+
+import { validateToken } from 'lib/api-validate-token'
+import { getAllCollections } from 'lib/learn-client/api/collection'
+import {
+	Collection as ClientCollection,
+	TutorialLite as ClientTutorialLite,
+} from 'lib/learn-client/types'
+import { activeProductSlugs } from 'lib/products'
+import { SectionOption } from 'lib/learn-client/types'
+import { getTutorialSlug } from 'views/collection-view/helpers'
+import { getCollectionSlug } from 'views/collection-view/helpers'
+import { ProductSlug } from 'types/products'
+
+const BATCH_SIZE = 10
+const DELAY_TIME = 200 // ms
+
+/**
+ * Accepts a POST request, triggers revalidation for all tutorial paths for all products
+ * landing pages, collection pages, and tutorial pages associated
+ */
+async function handler(request: NextApiRequest, response: NextApiResponse) {
+	if (request.method !== 'POST') {
+		response.status(StatusCodes.NOT_FOUND)
+		return
+	}
+
+	try {
+		const tutorialLandingPaths = getTutorialLandingPaths()
+		const collectionAndTutorialPaths = await getCollectionAndTutorialPaths()
+		const paths: string[] = [
+			...tutorialLandingPaths,
+			...collectionAndTutorialPaths,
+		]
+
+		if (!paths || paths.length === 0) {
+			response.status(200).end()
+		}
+
+		// Loop over all paths to revalidate in batches
+		// as this endpoint will fire off >1000 revalidation requests
+		let batchRevalidatePromises = []
+		for (let i = 0; i < paths.length; i++) {
+			// remove any trailing slash
+			const path = paths[i].replace(/\/$/, '')
+			console.log('[Revalidate]', path)
```

Comment:
nit, for consistency:
```suggestion
			console.log('[revalidate]', path)
```